### PR TITLE
Added a rolling window counter

### PR DESCRIPTION
**Pull request by Alars-ALIT**
A suggestion to fix issue #24

### CODE REVIEW

File: src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerImpl.java [hunk 1]
```diff
@@ -416,4 +422,38 @@ public CircuitBreakerMetrics getMetrics() {
   public CircuitBreakerOptions options() {
     return options;
   }
+  
+  public static class RollingCounter {		
+    private Map<Long, Long> window;
+    private long timeUnitsInWindow;
+	private TimeUnit windowTimeUnit;
```
Comment (cescoffier) [inline src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerImpl.java]: indetnetation should be fixed

File: src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerImpl.java [hunk 2]
```diff
@@ -416,4 +422,38 @@ public CircuitBreakerMetrics getMetrics() {
...
+    
+    public RollingCounter(long timeUnitsInWindow, TimeUnit windowTimeUnit) {
+        this.windowTimeUnit = windowTimeUnit;
+		this.window = new LinkedHashMap<>((int)timeUnitsInWindow + 1);
```
Comment (cescoffier) [inline src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerImpl.java]: indentation

**Comment by cescoffier** [comment]
Except for a few indentation issues, it looks good to me.

**Comment by cescoffier** [comment]
Looks good!, did you sign the Eclipse CLA?

**Comment by Alars-ALIT** [comment]
Nice! Just signed the CLA.

**Comment by cescoffier** [comment]
Thanks!

**Pull request merged**